### PR TITLE
Improve multi cluster hub error

### LINKED ISSUE #192: Improve errors from following the user guide with a multi-cluster environment.
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:
The user guide lists the steps as:
* init the cluster
* build images
* apply cluster (init sets the current cluster to the name of the cluster - not the hub)

If you choose a multi cluster setup then when I run the apply step I get an error about missing variables, some examples:

```
DEBU[0031] * module.kubernetes.var.vault_url: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'vault_vault_url' for variable 'data.terraform_remote_state.hub_state.v
ault_vault_url'  app=tarmak module=terraform std=err                                                                                                                                          
DEBU[0031] * module.kubernetes.var.vault_ca: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'vault_vault_ca' for variable 'data.terraform_remote_state.hub_state.vau
lt_vault_ca'  app=tarmak module=terraform std=err                                                                                                                                             
DEBU[0031] * module.kubernetes.var.public_zone_id: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'state_public_zone_id' for variable 'data.terraform_remote_state.h
ub_state.state_public_zone_id'  app=tarmak module=terraform std=err                                                                                                                           
DEBU[0031] * output.bastion_instance_id: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'bastion_bastion_instance_id' for variable 'data.terraform_remote_state.hub_
state.bastion_bastion_instance_id'  app=tarmak module=terraform std=err                                                                                                                       
DEBU[0031] * module.kubernetes.var.private_zone: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'network_private_zone' for variable 'data.terraform_remote_state.hub
_state.network_private_zone'  app=tarmak module=terraform std=err                                                                                                                             
DEBU[0031] * output.instance_fqdns: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'vault_instance_fqdns' for variable 'data.terraform_remote_state.hub_state.vault_
instance_fqdns'  app=tarmak module=terraform std=err                                                                                                                                          
DEBU[0031] * module.kubernetes.var.bastion_security_group_id: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'bastion_bastion_security_group_id' for variable 'data.
terraform_remote_state.hub_state.bastion_bastion_security_group_id'  app=tarmak module=terraform std=err                                                                                      
DEBU[0031] * module.kubernetes.var.vpc_id: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'network_vpc_id' for variable 'data.terraform_remote_state.hub_state.netwo
rk_vpc_id'  app=tarmak module=terraform std=err  
```

It's not immediately obvious that this is caused by the hub having not being applied - certainly there's no reference to this in the User Guide we have for AWS at the moment.

This can be fixed changing the current cluster in the tarma config to env_name-hub; applying the hub; changing it back and applying the cluster.

**What you expected to happen**:

I think we can do a few things of varying difficulty here.

1. Add a note in the user guide explaining that the hub must be applied first
2. Give a better error when the hub variables are missing.
3. Auto-apply the hub if it's missing?

**How to reproduce it (as minimally and precisely as possible)**:
Follow the user guide; create a multi-cluster env; don't apply the hub first.

### PR DESCRIPTION
**Pull request by JoshVanL**
**What this PR does / why we need it**:
Adds a verification to ensure that there is a remote state of the hub modules when applying a multi-cluster. This also adds some information in the docs.

fixes #192

/cc @charlieegan3 

**Release note**:

```release-note
Verify the hub cluster has been applied before a multi-cluster has
```

### CODE REVIEW

**Comment by JoshVanL** [comment]
@charlieegan3 
Yes I believe so. I wasn't able to do it without the specific ../s3manager downloader anyway.

**Comment by charlieegan3** [comment]
I get errors like this:

```
tarmak|./tarmak_local_build clusters apply 
2018/06/27 15:57:34 [TRACE] Preserving existing state lineage "f002a071-4af5-b799-965b-72e7a0fda458"
2018/06/27 15:57:34 [TRACE] Preserving existing state lineage "f002a071-4af5-b799-965b-72e7a0fda458"
FATA[0007] Tarmak exited with an error: failed to validate tarmak cluster: 1 error occurred:

* 2 errors occurred:

* remote state module 'bastion' is empty
* remote state module 'vault' is empty

Has the hub cluster been applied?
  error="failed to validate tarmak cluster: 1 error occurred:\n\n* 2 errors occurred:\n\n* remote state module 'bastion' is empty\n* remote state module 'vault' is empty\n\nHas the hub cluster been applied?\n"
```

Using the build from master I get the expected less helpful errors:
```
DEBU[0032] * module.kubernetes.var.bastion_security_group_id: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'bastion_bastion_security_group_id' for variable 'data.terraform_remote_state.hub_state.bastion_bastion_security_group_id'  app=tarmak module=terraform std=err
DEBU[0032] * module.kubernetes.var.private_zone_id: Resource 'data.terraform_remote_state.hub_state' does not have attribute 'network_private_zone_id' for variable 'data.terraform_remote_state.hub_state.network_private_zone_id'  app=tarmak module=terraform std=err
DEBU[0032]                                               app=tarmak module=terraform std=err
DEBU[0032]                                   app=tarmak module=terraform std=err
ERRO[0032] Tarmak exited with an error: exit status 1    error="exit status 1"
```

So that seems to work as far as I can tell!

**Comment by charlieegan3** [comment]
@JoshVanL I'll assign Christian to get an `/approve`

/assign @simonswine 
/unassign

File: pkg/tarmak/provider/amazon/tf_remote_state.go [hunk 1]
```diff
@@ -216,3 +218,25 @@ func (a *Amazon) validateRemoteStateDynamoDB() error {
 
 	return nil
 }
+
+func (a *Amazon) RemoteStateData(path string) (*terraform.State, error) {
```
Comment (simonswine) [inline pkg/tarmak/provider/amazon/tf_remote_state.go]: I would be prefer using here the logic coming from `terrafrom output`. That would make us more independet from cloud provider

File: pkg/tarmak/cluster/cluster.go [hunk 2]
```diff
@@ -157,8 +157,43 @@ func (c *Cluster) validateInstancePools() (result error) {
...
+		if err := c.verifyHubState(); err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+
+	return result.ErrorOrNil()
+}
+
+func (c *Cluster) verifyHubState() error {
+	errMsg := "\n\nHas the hub cluster been applied?\n"
```
Comment (simonswine) [inline pkg/tarmak/cluster/cluster.go]: Not too sure why this is a question and with new lines

**Comment by JoshVanL** [comment]
I've changed over to using `terraform output`, what do you think?

/unassign
/assign @simonswine

**Pull request merged**